packages/gatsby-plugin-feed/README.md

```diff
@@ -78,10 +78,11 @@ Each feed must include `output`, `query`, and `title`. Additionally, it is stron
 
 `match` is an optional configuration, indicating which pages will have feed reference included. The accepted types of `match` are `string` or `undefined`. By default, when `match` is not configured, all pages will have feed reference inserted. If `string` is provided, it will be used to build a RegExp and then to test whether `pathname` of current page satisifed this regular expression. Only pages that satisifed this rule will have feed reference included.
 
-All additional options are passed through to the [`rss`][rss] utillity. For more info on those additional options, [explore the `itemOptions` section of the `rss` package](https://www.npmjs.com/package/rss#itemoptions).
+All additional options are merged with `query.site.siteMetadata`, and processed through `setup` option (which is a function). Return value of `setup` is, in turn, passed as a [feedOptions][rss-feedOptions] argument of the [`rss`][rss] utillity. For instance, you can return object containing `site_url` field from `setup` function to set the top-level `<link>` tag of the feed. For more info on those additional options, [explore the `itemOptions` section of the `rss` package](https://www.npmjs.com/package/rss#itemoptions).
```

Comment:
Documentation is a tricky thing. While we don't want others to go through what you experienced in #14136, I also don't necessarily think providing this would clarify to others exactly how this should be used.

Additionally -- we want to keep our documentation concise and approachable, and this change seems to lessen that and could introduce confusion as to _why_ we're telling someone this info they may not need.

Reply:
It seems like we need an "advanced usage" section possibly. I'm still not quite grokking what the problem was here, though. I'll ask in #14136.